### PR TITLE
Convert map.py transforms over to base map.

### PR DESCRIPTION
**Pull request by eric-anderson**
* Remove some unused generate_* functions
* Adjust transforms to import the generate function from the original source rather than map since map no longer uses them. Ideally we'll remove all ray specific bits from the transforms when converting them over.

### CODE REVIEW

File: lib/sycamore/sycamore/transforms/map.py [hunk 1]
```diff
@@ -62,20 +63,39 @@ def custom_flat_mapping_function(document: Document) -> list[Document]:
...
+            class _Wrap(f):  # type: ignore[valid-type,misc]
+                def __init__(self, *args, **kwargs):
+                    super().__init__(*args, **kwargs)
+
+                def __call__(self, docs):
+                    s = super()
+                    ret = []
+                    for d in docs:
+                        ret.extend(s.__call__(d))
+                    return ret
```
Comment (HenryL27) [inline lib/sycamore/sycamore/transforms/map.py:81]: potential one-liner alternatives:
```python
return sum((s.__call__(d) for d in docs), start=[])

return list(itertools.chain.from_iterable(s.__call__(d) for d in docs)))
```

File: lib/sycamore/sycamore/transforms/map.py [hunk 2]
```diff
@@ -62,20 +63,39 @@ def custom_flat_mapping_function(document: Document) -> list[Document]:
...
+            return _Wrap
         else:
-            ray_callable = generate_flat_map_function(self._f)
-            return input_dataset.flat_map(ray_callable, **self.resource_args)
 
+            def _wrap(docs):
+                assert isinstance(docs, list)
+                ret = []
+                for d in docs:
+                    assert isinstance(d, Document)
```
Comment (HenryL27) [inline lib/sycamore/sycamore/transforms/map.py:90]: should these asserts also be in the class part?

File: lib/sycamore/sycamore/transforms/map.py (hunk 1 above)
Comment (HenryL27) [reply lib/sycamore/sycamore/transforms/map.py:81]: nvm, seems like the manual loop is actually faster

File: lib/sycamore/sycamore/transforms/map.py [hunk 3]
```diff
@@ -102,20 +122,12 @@ def __init__(
...
-    def execute(self) -> "Dataset":
-        input_dataset = self.child().execute()
-        if isinstance(self._f, type):
-            ray_callable = generate_map_batch_class(
-                self._f, self._f_args, self._f_kwargs, self._f_constructor_args, self._f_constructor_kwargs
-            )
-            return input_dataset.map_batches(ray_callable, compute=ActorPoolStrategy(size=1), **self.resource_args)
-        else:
-            ray_callable = generate_map_batch_function(self._f)
-            return input_dataset.map_batches(ray_callable, **self.resource_args)
```
Comment (HenryL27) [inline lib/sycamore/sycamore/transforms/map.py:121]: where did this go?

**Review by HenryL27 — COMMENTED** [review]
mostly nits, main question is where did the ray dataset calls end up? Maybe they got moved in the prev PR?

File: lib/sycamore/sycamore/transforms/map.py (hunk 3 above)
Comment (eric-anderson) [reply lib/sycamore/sycamore/transforms/map.py:121]: All the interaction with ray happens in BaseMapTransform.

**Review by HenryL27 — APPROVED** [review]

File: lib/sycamore/sycamore/transforms/base.py [hunk 4]
```diff
@@ -37,21 +37,23 @@ def __init__(
         kwargs: Optional[dict[str, Any]] = None,
         constructor_args: Optional[Iterable[Any]] = None,
         constructor_kwargs: Optional[dict[str, Any]] = None,
+        # If we auto-generate lineage, then the conversion to BaseMap has to go in a single PR
+        # since everything needs to be updated to skip metadata. If we temporarily disable the
+        # lineage metadata, then we can do the conversion to BaseMap in separate PRs.
+        enable_auto_metadata=False,
```
[the remark this answers is not in the document]
Comment (eric-anderson) [reply lib/sycamore/sycamore/transforms/base.py:43]: Done.

File: lib/sycamore/sycamore/transforms/base.py [hunk 5]
```diff
@@ -61,15 +63,37 @@ def execute(self) -> "Dataset":
...
 
+    def _local_process(self, in_docs: list[Document]) -> list[Document]:
+        """Internal function for faster testing during the conversion to running on BaseMap.
+        If extended with metadata support, this could become more real."""
+        import copy
+
+        # transforms assume they can mutate docs in place; this works in ray because documents are serialized and
+        # deserialized between every stage.
+        docs = copy.deepcopy(in_docs)
+        if isinstance(self._f, type):
```
[the remark this answers is not in the document]
Comment (eric-anderson) [reply lib/sycamore/sycamore/transforms/base.py:74]: Added comment.

File: lib/sycamore/sycamore/transforms/map.py (hunk 1 above)
Comment (eric-anderson) [reply lib/sycamore/sycamore/transforms/map.py:81]: Ack.

File: lib/sycamore/sycamore/transforms/map.py (hunk 2 above)
Comment (eric-anderson) [reply lib/sycamore/sycamore/transforms/map.py:90]: They were for helping me debug problems, but don't hurt to put in both branches, so I did.

File: lib/sycamore/sycamore/transforms/map.py (hunk 3 above)
Comment (eric-anderson) [reply lib/sycamore/sycamore/transforms/map.py:121]: The same as the code here?
https://github.com/aryn-ai/sycamore/blob/7c7d3ce4aa48bd1543c70af20a9056f87c52e252/lib/sycamore/sycamore/transforms/base.py#L59

Comment (HenryL27) [reply lib/sycamore/sycamore/transforms/map.py:121]: yeah, that! where's that thing you linked at me right there ;)
I guess I was blinded by the flashy _process_ray method

**Pull request merged**